bftengine/src/communication/TlsTCPCommunication.cpp

```diff
@@ -326,32 +288,35 @@ class AsyncTlsConnection : public enable_shared_from_this<AsyncTlsConnection> {
                          string connectionType,
                          string subject,
                          NodeNum expectedPeerId = -1) {
+
+    // first, basic sanity test, just to eliminate disk read if the certificate
+    // is unknown
     regex r("OU=\\d*", regex_constants::icase);
     smatch sm;
     regex_search(subject, sm, r);
     if (4 > sm.length()) {
-      LOG_ERROR("OU not found or empty: " << subject);
+      LOG_ERROR(_logger, "OU not found or empty: " << subject);
       return false;
     }
 
-    string remPeer = sm.str().substr(0, sm.str().length() - 3);
+    string remPeer = sm.str().substr(3, sm.str().length() - 3);
     if (0 == remPeer.length()) {
-      LOG_ERROR("OU empty " << subject);
+      LOG_ERROR(_logger, "OU empty " << subject);
       return false;
     }
 
     int remotePeerId;
     try {
-      remotePeerId = stoi(sm.str());
+      remotePeerId = stoi(remPeer);
     } catch (const std::invalid_argument &ia) {
```

Comment:
Correct me if there is some nuance here I am missing (such as this catch being required for the code to compile, if that is the case), but the fact we are catching an `std::invalid_argument` here for this `stoi` call suggests we do not trust that `remPeer` is a valid string representation of an `int`. If we do distrust the contents of `remPeer`, should we not also be catching an `std::out_of_range`, as I believe `stoi` can also throw one of those?

Reply:
Thanks, good point. fixed.